test/parallel/test-http-readable-data-event.js

```diff
@@ -27,14 +32,15 @@ const server = http.createServer((req, res) => {
   const expectedRead = [helloWorld, null, helloAgainLater, null];
 
   const req = http.request(opts, (res) => {
-    res.on('error', common.mustNotCall);
+    res.on('error', common.mustNotCall());
 
     res.on('readable', common.mustCall(() => {
       let data;
 
       do {
         data = res.read();
         assert.strictEqual(data, expectedRead.shift());
+        next();
```

Comment:
I suggest to remove the `next` and use a boolean to verify that `res.end(helloAgainLater)` was not yet called. That should be simpler but does the same as far as I read the code.

```js
// Here
if (!streamEnded) {
  res.end(helloAgainLater);
  streamEnded = true;
}
// At the top of the file
let streamEnded = false;
```

Reply:
The `next` call can be moved after the loop but it's also ok as is.